Sources/BaggageContext/BaggageContext.swift

```diff
@@ -0,0 +1,59 @@
+/// A `BaggageContext` is a heterogenous storage type with value semantics for keyed values in a type-safe
+/// fashion. Its keys are uniquely identified via `BaggageContextKey`s. These keys also dictate the type of
+/// value allowed for a specific key-value pair through their associated type `Value`.
+///
+/// # Subscript access
+/// You may access the stored values by subscripting with a key type conforming to `BaggageContextKey`.
+///
+///     enum MyContextKey: BaggageContextKey {
+///       typealias Value = String
+///     }
+///
+///     var baggage = BaggageContext()
+///     // set a new value
+///     baggage[MyContextKey.self] = "abc"
+///     // retrieve a stored value
+///     baggage[MyContextKey.self] ?? "default"
+///     // remove a stored value
+///     baggage[MyCoolTraceID.self] = nil
+///
+/// Libraries may also want to provide an extension, offering the values that users are expected to reach for
+/// using the following pattern:
+///
+///     extension BaggageContext {
+///       var myCoolTraceID: String {
```

Comment:
```suggestion
///       var myContextKey.self: String {
```

same name but lowercase as the Key type

Reply:
Isn't context key and trace ID something different? I thought that `TraceID` would e.g. be a struct that's used as the associated value of a `TraceIDKey` enum.

Reply:
To make it more obvious, we could also write `var myCoolTraceID: MyContextKey.Value`.

Reply:
> Isn't context key and trace ID something different? I thought that TraceID would e.g. be a struct that's used as the associated value of a TraceIDKey enum.

Right. Maybe I'm missing something?

The key is `MyContextKey`, the value is `MyContext.Value`, the extension would be `myContext: MyContext.Value`

A more real example would be:

- `ZipkinTraceIDKey`
- `ZipkinTraceIDKey.Value` == `ZipkinTraceID`
- the var `zipkinTraceID: ZipkinTraceIDKey.Value`

right?

> we could also write var myCoolTraceID: MyContextKey.Value.

Yeah that's good 👍

Reply:
Okay, then I'd suggest using
- `MyTraceID`
- `MyTraceIDKey`
- `MyTraceIDKey.Value` == `MyTraceID`
- `myTraceID: MyTraceIDKey.Value`

Or do you think it makes more sense to not be tracing-specific here?

Reply:
And in `BaggageContextTests` I'd also rename `TestContextKey` to `TestIDKey` and create a property `testID` to the `BaggageContext`.